runtime/ast/expression.go

```diff
@@ -220,6 +220,68 @@ func (*StringExpression) precedence() precedence {
 	return precedenceLiteral
 }
 
+// StringTemplateExpression
+
+type StringTemplateExpression struct {
+	Values      []string
+	Expressions []Expression
+	Range
+}
+
+var _ Expression = &StringTemplateExpression{}
+
+func NewStringTemplateExpression(
+	gauge common.MemoryGauge,
+	values []string,
+	exprs []Expression,
+	exprRange Range,
+) *StringTemplateExpression {
+	common.UseMemory(gauge, common.NewStringTemplateExpressionMemoryUsage(len(values)+len(exprs)))
+	return &StringTemplateExpression{
+		Values:      values,
+		Expressions: exprs,
+		Range:       exprRange,
+	}
+}
+
+var _ Element = &StringExpression{}
+var _ Expression = &StringExpression{}
+
+func (*StringTemplateExpression) ElementType() ElementType {
+	return ElementTypeStringTemplateExpression
+}
+
+func (*StringTemplateExpression) isExpression() {}
+
+func (*StringTemplateExpression) isIfStatementTest() {}
+
+func (*StringTemplateExpression) Walk(_ func(Element)) {
+	// NO-OP
+}
+
+func (e *StringTemplateExpression) String() string {
+	return Prettier(e)
+}
+
+func (e *StringTemplateExpression) Doc() prettier.Doc {
+	return prettier.Text(QuoteString("String template"))
```

Comment:
This needs to reproduce the expression as a literal. If this is not supported yet as part of this PR, maybe just leave a `// TODO` and `panic("not implemented")`

Reply:
Left it as a `//TODO` with `panic` for this PR.